src/main/io/gps.c

```diff
@@ -688,79 +1085,156 @@ void gpsInitUblox(void)
+                            break;
                         }
+                        ubloxSetMessageRate(CLASS_NAV, MSG_NAV_SAT, 5); // set SAT MSG rate (every 5 cycles)
                         break;
                     case UBLOX_MSG_SATINFO:
-                        if (gpsData.ubloxUseSAT) {
-                            ubloxSetMessageRate(CLASS_NAV, MSG_SAT, 5); // set SAT MSG rate (every 5 cycles)
-                        } else {
-                            ubloxSetMessageRate(CLASS_NAV, MSG_SVINFO, 5); // set SVINFO MSG rate (every 5 cycles)
+                        if (gpsData.unitVersion > UBX_VERSION_M8) {
+                            gpsData.state_position++;
+                            break;
                         }
+                        ubloxSetMessageRate(CLASS_NAV, MSG_NAV_SVINFO, 5); // set SVINFO MSG rate (every 5 cycles)
                         break;
                     case UBLOX_MSG_DOP:
-                        ubloxSetMessageRate(CLASS_NAV, MSG_DOP, 1); // set DOP MSG rate
+                        if (ubloxHasValSetGet()) {
+                            ubloxSetMessageRateValSet(CFG_MSGOUT_UBX_NAV_DOP_I2C, 0);
+                            ubloxSetMessageRateValSet(CFG_MSGOUT_UBX_NAV_DOP_SPI, 0);
+                            ubloxSetMessageRateValSet(CFG_MSGOUT_UBX_NAV_DOP_UART1, 1);
+                            break;
+                        }
+                        ubloxSetMessageRate(CLASS_NAV, MSG_NAV_DOP, 1); // set DOP MSG rate
                         break;
                     case UBLOX_SET_NAV_RATE:
-                        ubloxSetNavRate(gpsConfig()->gps_update_rate_hz, 1, 1);
+                        ubloxSetNavRate(gpsData.updateRate, 1, 1);
                         break;
                     case UBLOX_SET_SBAS:
                         ubloxSetSbas();
                         break;
                     case UBLOX_SET_PMS:
+                        if (gpsData.unitVersion != UBX_VERSION_M8) {
+                            gpsData.state_position++;
+                            break;
+                        }
                         ubloxSendPowerMode();
                         break;
                     case UBLOX_MSG_CFG_GNSS:
                         if ((gpsConfig()->sbasMode == SBAS_NONE) || (gpsConfig()->gps_ublox_use_galileo)) {
-                            ubloxSendPollMessage(MSG_CFG_GNSS);
+                            if (false && ubloxHasValSetGet()) {
+                                uint8_t payload[4];
+                                memset(payload, 0, 4);
+                                ubxMessage_t tx_buffer;
+                                size_t length = ubloxValSet(&tx_buffer, (uint32_t)CFG_SIGNAL_GPS_ENA, payload, UBX_VAL_LAYER_RAM); //0x1031001f
+                                tx_buffer.payload.cfg_valset.version = 1;
+                                /*length += ubloxValGet(&tx_buffer, CFG_SIGNAL_SBAS_ENA, length);
```

Comment:
Is this PR ready for review? Many dead / commented out code

Reply:
Dead/commented out code may, or may not, be kept, depending on whether it is useful perhaps in future.

For example, there is some kind of complex logging code left over from Tony Blit's work, which sends values to an OLED.  I intend to comment all that stuff out, rather than delete it, in case someone wants to restore it for some purpose.

Most temporary commented out code will be removed

Reply:
I see your good intentions, and I agree, it's good to preserve knowledge.
At the same time, commented out code should never be kept and typically frowned upon by coding guidelines - https://codeql.github.com/codeql-query-help/cpp/cpp-commented-out-code

If you strongly feel previous code should be referred to, then it's the role of source control to keep that history, and your job to point to that history, with something like this

```
// If you want to do X, a sample code here https://github.com/betaflight/betaflight/blob/90e792e2438344a2b05b377c2f1925aa9829fadd/src/main/io/gps.c#L811
```

Otherwise, for the rest of us, these commented out blocks are confusing

Reply:
Comment noted